scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/macros/Macros.scala

```diff
@@ -38,17 +36,14 @@ object Macros {
   def caseClassParquetSchema[T]: String = macro ParquetSchemaProvider.toParquetSchemaImpl[T]
 
   /**
-   * Macro used to generate parquet tuple converter for a given case class.
-   *
-   * @tparam T Case class type that contains primitive or collection type fields or nested case class.
-   * @return Generated parquet converter
+   * Macro generated case class read support
    */
-  def caseClassParquetTupleConverter[T]: ParquetTupleConverter[T] = macro ParquetTupleConverterProvider.toParquetTupleConverterImpl[T]
+  def caseClassParquetReadSupport[T](schema: String): ParquetReadSupport[T] = macro ParquetReadSupportProvider.toParquetReadSupportImpl[T]
```

Comment:
why is the schema here? Why not call the toParquetSchemaImpl macro? Seems it just requires extra work for the user.

Can this be implicit?